### ISSUE TITLE
Cron treated Dailies as unchecked / Quest damage ran twice - doubled boss/collection quest progress

### ISSUE THREAD

**Issue by brightbold** [HabitRPG/habitica#2805]
So my quest damage ran twice today:
![image](https://f.cloud.github.com/assets/405068/2120045/7bdbdc68-919c-11e3-8ba3-16498ffc5e5b.png)

I do keep two Habit tabs open (one for chat and one for tasks) so that's possibly a culprit? But I've been doing that for a while and this has never happened before. Also, I am using the beta site, but @paglias didn't think the new beta code was at fault.

---

_edit by admin:_
## If you receive this bug, please comment here and tell us:
- **Do you use Habitrpg on more than one device? If so, are the time and timezone settings different between them?**
- **Is the damage identical, or different?**
- **Please post a screenshot of your party chat where it shows the damage being done to the boss and your party. (Do this by logging into Github and dragging your screenshot onto a comment; emailing images won't work.)**

You and your party members can recover from the additional damage received by going to [Settings -> Site -> Fix Character Values](https://habitrpg.com/#/options/settings/settings) to adjust your health, and your level, gold, and experience if necessary.
##

**Comment by deilann** [comment]
Had this happen to one of my party members a couple quests ago. && damage stayed registered; Gryphon definitely took the hit twice.

**Comment by lefnire** [comment]
Definitely a result of the duplicate ops GH issue floating around, we'll be hitting that hard in coming weeks

**Comment by MtnLioness** [comment]
Ah, that may be what happened to me too.

**Comment by hattieshoe** [comment]
This has happened to me several times in the past, and again today. 
![image](https://cloud.githubusercontent.com/assets/5732647/2993266/a4a2e392-dcad-11e3-8743-17bfe8d76db3.png)
The Ghost Stag took the double damage. Not that I'm complaining much about that, but it makes me nervous about the previous few days where I missed a lot of dailies -- if this had happened then and if the party takes their quest damage twice as well, I would have killed my whole party.

**Comment by Alys** [comment]
Also happened to @georgit and @SabreCat in https://github.com/HabitRPG/habitrpg/issues/3893

and to Emocube (6a962f71-8b15-44cb-bcfa-107b9ab01c15) in the Tavern. [Screenshot](https://farm4.staticflickr.com/3892/14876303679_3d4b3b8728_o.png)

**Comment by Alys** [comment]
More reports in https://github.com/HabitRPG/habitrpg/issues/4038 which seem to indicate that it happens repeatedly for some people, while not happening at all for others in the same party.

**Comment by me-and** [comment]
Following up from the notes in #4038, this is still happening in my party on the new quest we've just started.

**Comment by me-and** [comment]
Oh, more relevant points from me: I frequently have multiple windows open, such as my pet page and my inventory so I can see what I've already hatched before I hatch my next pet, or my to-do list on my home and work PCs simultaneously.

I have never used the beta site, so that's not the culprit in my case.

**Comment by brightbold** [comment]
Has anyone noticed a correlation between the double boss damage and dailies not initially resetting #1057? Both happened to me yesterday -- I opened Habit and it still showed all the dailies checked off from the day before; I refreshed the page a couple of times and got dailies to reset for the new day. Then when I checked my party page, I saw the boss damage had been applied twice. 

I have the idea that maybe this is not the first time both have happened to me on the same day. Issue #3893, cited above, seems to indicate that the act of refreshing the page at the wrong time could cause cron to run twice. So I think we should pursue @georgit's theories on refreshing from that duplicate issue.

**Comment by me-and** [comment]
I've never had a problem with dailies not resetting, but I have repeated problems with double boss damage.

**Comment by polishsputnik** [comment]
This happened to my party a couple of days ago. 09fb4068-321a-42e6-8687-3f51671fd195 is the group id. Working on getting the user ID of the person it happened to. This is how they describe it: I went to Social:Party first and hit refresh which caused the quest to update with damage. When I then went to my tasks and ticked off my first one it refreshed again, doubling the damage for the day

**Comment by polishsputnik** [comment]
BTW, this happened to user 63c85a7d-aa6e-4ea6-9a22-4a663fe78aa3 in Google Chrome.

**Comment by clmmerrill** [comment]
Hi our party just had the double damage happen. Quest quest quest. I don't mind dying but my Aurora Hat Broke.  
chupipe attacks Rat King for 2.3 damage, Rat King attacks party for 44.4 damage. -2 hours ago+1  
chupipe attacks Rat King for 2.3 damage, Rat King attacks party for 44.4 damage. -2 hours ago+1

**Comment by Alys** [comment]
@clmmerrill On the website, go to Settings -> Site and click the Fix Character Values button. On the screen that appears, you can increase your level by 1 and restore your lost gold plus the extra required to buy the hat.

**Comment by clmmerrill** [comment]
@Alys  thanks. I don't really need to replace gold since I have tons. I will send email if I can't figure out how to buy the hat again.

**Comment by Alys** [comment]
In my non-moderator account (3e595299-3d8a-4a10-bfe0-88f555e4aa0c) I got the double cron bug this morning. The party chat reported the same message twice for damage I did to the boss, and the damage was definitely applied to the boss twice, because the boss died instead of being at the expected half total health. My HP change shows that the damage from my Dailies and from the boss was applied to me twice.

My MP last night after finishing all actions on the site was 116, and this morning after cron was 146 (maximum 302) so it looks like my MP was increased only once, not twice. 

All of my Dailies were reset correctly.

**Comment by Alys** [comment]
AmandaMock has reported this a second time (see the two issues listed above). It seems to happen to them much more often than to anyone else.

**Comment by Alys** [comment]
AmandaMock has left HabitRPG (https://github.com/HabitRPG/habitrpg/issues/4745#issuecomment-76019544).

**Comment by brightbold** [comment]
@Alys I saw your comment to AmandaMock:

> It might be possible that if a player had HabitRPG open in two tabs at once and then reloaded both of them at exactly the same time first thing in the morning, then that would trigger cron twice, but it really would have to be a simultaneous reload in both tabs, which would be very difficult to get the timing right for...

and I wondered whether you should add the "do you keep Habit open in two tabs" question to your troubleshooting info requests above. As I mentioned in the OP I do keep two tabs open (one for tasks, and one for everything else, so that I can always go straight to the tasks tab to be productive without getting distracted by guilds & equipment), and I've always been a little suspicious that that may be the cause of this happening to me. (It's happened more than the two times I've commented here, but still infrequently, and not since I last commented on it in October.) It might be a coincidence, but seems worth collecting data on.

And another data point - if this happens on the day you complete a quest, you get double eggs too. So sometimes you're rewarded for your troubles!

**Comment by crookedneighbor** [comment]
Happened to me last night. I did not have two tabs open.

![screen shot 2015-02-26 at 8 28 19 am](https://cloud.githubusercontent.com/assets/2916945/6393660/a939d346-bd91-11e4-917d-95069dc367d0.png)

**Comment by Alys** [comment]
https://github.com/HabitRPG/habitrpg/issues/4174#issuecomment-76521621 and the comment below it describe an exceptional example of duplicate quest damage. The duplicate damage was enough to supposedly kill the boss, causing the participants' accounts to be treated as if the quest had ended. However the duplicate damage was NOT actually applied twice to the quest boss's health, so the boss did not actually die. The result was that participants received the quest rewards and achievement and their own user.party.quest objects were cleared out, but the party's quest object was not cleared out and the boss was not dead (leaving the quest in a state where it could never be ended).

**Comment by pjay-io** [comment]
@Alys Couldn't duplicate damage easily be avoided by registering the cycle (times of crons passed) per user and setting a unique index for uid quest_id croncycle ? Or at least checking these values for damage already existing in a pair like that.

**Comment by pjay-io** [comment]
As far as I read all the issues mentioned above, it is really an issue with identifying if a cron has already run that day isn't it?

What is the exact reasoning of choosing triggering cron when logging in versus. server running cron?

**Comment by Alys** [comment]
@PJay- It might not be cron running twice - or might not always be caused by that. My comment at https://github.com/HabitRPG/habitrpg/issues/2805#issuecomment-69270732 was one example where quest progress was applied twice, but my MP was not incremented twice, so it sort of looks like cron did not run twice - just that the quest progress happened twice. However, I'm not against us investigating ways of preventing cron running twice within the same day. :)  I'm not sure about login-vs-server cron - load issues I think. There have been discussions here about cron - probably [somewhere in this lot](https://github.com/HabitRPG/habitrpg/issues?q=label%3Acron), although I know that doesn't narrow it down much.

**Comment by pjay-io** [comment]
@Alys Yeah I see what you mentioned, if the poormanscron implementation is well enough done to prevent double cron's by somehow knowing if it is already done for that cycle, that would be fine I guess. 

Other things are also buggy because of that like me changing the CDS after 00:00 to 03:00 ran the cron again at 03:00, only 3 hours after it. That should be prevented.

Anyway I agree with you on boss damage, and I think it should be a relatively easy fix. Something like:
- Check last time cron has been run for user
- Compare that with last time boss damage dealt
- If those are the same, boss damage already dealt, abort abort

I would be a proponent of using id's or counters for cron though instead of time (heavy load, time differs between the two)

**Comment by eiron** [comment]
Ah yes. It is not "multi-cronning" as the MP only goes up the once. It is only the damage given to and from bosses, and quest pick-ups. Still, it does make it rather frightening if I have incomplete dailies and the boss is hence going to strike an unknown number of times.

**Comment by Memry** [comment]
I had it happen my last two quests.  We just started another quest.  I'll see if it happens again.  I have two habit accounts, and it only happens on one account.  The account it does NOT happen to is the one I use on two devices (using the habit app on my iPad and habit in IE on my home computer).  The other account I only use in Chrome on my home computer.

**Comment by littlezildjian** [comment]
I use Habit on my browser (chrome) and my mobile app. I do often have both chrome and the app open simultaneously on my phone (I use the app mostly, but keep chrome so I can use abilities). So at most it's open in 3 places.

The damage is always identical and does take effect twice (both for attack and damage to party). I'm doing the Recidivate Transformed quest, and this is the first time it's ever happened.

If it helps, my userID is: 5e728a42-1ce4-481e-9afc-8c202e467990 and my groupID is: c3062456-a967-4020-b56e-f7478deeebc3

![habitbug](https://cloud.githubusercontent.com/assets/9541071/7967968/e8fdd702-09ea-11e5-8122-e4b44f97f2c3.png)

**Comment by SabreCat** [comment]
I just had this happen to me this morning:

![image](https://cloud.githubusercontent.com/assets/4501321/7987926/3b19a938-0aa6-11e5-9cc7-70226b02f7cb.png)

I was trying to connect to one of the development servers, which returned a NO_RESPONSE error at first. So I opened a browser tab to connect to the production site, and in the meantime, the dev site loaded. When I made it to my party page, the damage had been applied twice.

**Comment by Alys** [comment]
@SabreCat Did you notice any other signs of cron running twice, such as gaining twice the mana overnight?

**Comment by SabreCat** [comment]
@Alys: Not that I recall, no, but it could well be that I didn't notice rather than that it didn't happen.

**Comment by gaballison** [comment]
I'm a rogue and this started happening to me two days ago on the [Wail of the Whale](http://habitica.wikia.com/wiki/Wail_of_the_Whale) quest:
![habit-double-01 2](https://cloud.githubusercontent.com/assets/12204283/8932796/49246400-350d-11e5-99f3-1d87d73a52cb.png)
I thought at first it was just a glitch in the chat but it happened again yesterday, which helped us to finish the quest sooner and now it happened again today while collecting fins and coral for the [Message in a Bottle](http://habitica.wikia.com/wiki/Message_in_a_Bottle) quest:
![habit-double-02](https://cloud.githubusercontent.com/assets/12204283/8932877/c9742870-350d-11e5-9873-171c4223e38a.png)
I access the site from both work and home using Chrome (just the site, not the HabitRPG extension) and both locations are in the same timezone. I often have two tabs open so I can feed and hatch pets, if that might be an issue. I am not using the beta site or developer site or anything like that, to my knowledge. I have these mobile apps installed on both my [iPad](https://itunes.apple.com/us/app/habitrpg/id689569235?mt=8) and [phone](https://play.google.com/store/apps/details?id=com.ocdevel.habitrpg&hl=en) but I rarely use them. My uID is f527a20c-8fb8-41b5-86c3-a606bd160d79 if that helps.

**Edit** I just remembered this: I usually leave the tab open when I let my laptop go to sleep at night, and at least yesterday, if not the day before, when I resumed using the site for the first time in the morning it looked like it hadn't run cron yet, even when I refreshed the page. I then clicked the manual sync button to reload so I could enter in my morning dailies, so I'm not sure if me manually syncing might be the problem.

**Comment by Alys** [comment]
@gaballison Thanks for all the details! I would be interested to know if it
still happens if you have only one tab open overnight.

For the collection quest items, does the count of total items found in the
quest box show the double progress (16 and 18 items, rather than 8 and 9)?

Have you noticed any other signs of cron running twice, such as twice as
much damage as you were expecting or twice as much mana gained?

**Comment by littlezildjian** [comment]
@Alys I am still having the same problem (original post June 3 on this thread) as @gaballison, and am doing a collection quest right now. For me it shows the progress twice (8 and 8) rather than double (16). In past boss quests, I do take twice the damage as well. I did also get a ton of Mana this morning...a lot more than usual...so it looks like I got Mana twice.

I typically have the app open all day on my phone, and the browser open all day on my computer. My computer gets shut down at night, but the app rarely does. If I want to feed my pets I will sometimes open a mobile browser as well, so maybe that's the issue? I rarely do a manual sync.

**Comment by gaballison** [comment]
@Alys I usually only keep one tab open overnight, not 2, so it may not be that after all. I just am usually too lazy to actually shut the whole machine down at night so I just close the lid and leave it be. :smile:

I admit I didn't pay close enough attention when we were doing a boss damage quest to see if it did do twice the damage, but I'm assuming it did because we finished the quest more quickly than I anticipated.

It definitely **is** counting my items twice, however. I'm in a party of 2 and we just started the Message in a Bottle quest yesterday, so our first items showed up today:
![habit-partydoubledmg-01](https://cloud.githubusercontent.com/assets/12204283/8937233/20fc5eac-3525-11e5-9494-e10b0ab3cbe4.png)
Instead of 14 Fire Coral (6+8) and 14 Blue Fins (5+9), we have 22 Fire Coral (6+8+8) and 23 Blue Fins (5+9+9):
![habit-partydoubledmg-02](https://cloud.githubusercontent.com/assets/12204283/8937245/28b2a412-3525-11e5-8371-2d9078573e08.png)

**Comment by gaballison** [comment]
Sorry to spam this thread but now it's counting my damage three times:
![habit-triple-edit](https://cloud.githubusercontent.com/assets/12204283/8993474/682620d8-36d3-11e5-9577-a24e4b941dae.png)
We've started a new quest and now have a third member with us so I wondered if that's why I'm now doing damage three times instead of just two. It is definitely applying my damage 3x, too. For what it's worth, I wasn't using the site or any app 10 hours ago (around 3:30/4 am) so I'm not sure why it ran then.

**Comment by brightbold** [comment]
I keep three tabs open in Chrome, which won't load them until you select the tab to make it active. The last time the double-damage bug hit me, I had clicked on one of my tabs, realized it wasn't the one I wanted, and immediately clicked on a second one. (Since then I've been careful to wait until the active tab has fully loaded before clicking on another, and I haven't had double damage in that time.) While that could have been a coincidence, it certainly seemed to me that I had triggered cron in two different locations almost simultaneously. I would guess you could do the same thing by simultaneously loading desktop and mobile.

**Comment by brandonlichtenwalner** [comment]
This happened to me yesterday and today. The only difference from my normal routine is that I logged in for the first time for the day from the Android app instead of through my web browser. Habitica was probably still open in a browser (with the computer in "sleep" mode) both times.

Screenshots:
![today](https://cloud.githubusercontent.com/assets/5702012/9493409/383ac6b2-4bcd-11e5-953e-5391feee66f8.PNG)
![yesterday](https://cloud.githubusercontent.com/assets/5702012/9493408/3838b3e0-4bcd-11e5-8efb-ac543ba68895.PNG)

User ID: ca0d3dee-4308-44b7-a0d2-045964ec7cb3

**Comment by AdrienLemaire** [comment]
I got the triple damage bug:

![a](https://cloud.githubusercontent.com/assets/260983/9830754/e18aefa8-5974-11e5-8e89-bd76511978fd.png)

I had only one tab open and only connected via the desktop version, but I went out of the inn (in which I stayed for almost a month). I also happened to came back from a trip overseas, maybe the timezone change had some effect.

**Comment by Alys** [comment]
There's another report of this in https://github.com/HabitRPG/habitrpg/issues/5975#issuecomment-143248445 for some of the party members in party 226e3812-5144-4e78-a2a4-d4040aed1af5. Damage log is below. The dates might be displayed in a different timezone than the party members are seeing. "Vice, the Shadow Worm" has been abbreviated to "Vice".

```
2015-09-24 07:54    Anna Kettner attacks Vice for 25.4 damage, Vice attacks party for 4.4 damage.
==========
2015-09-23 16:57    Sarah (Diyara) attacks Vice for 4.7 damage, Vice attacks party for 0.0 damage.
2015-09-23 15:48    naomiamerica attacks Vice for 8.9 damage, Vice attacks party for 0.0 damage.
2015-09-23 03:57    Anna Kettner attacks Vice for 0.0 damage, Vice attacks party for 49.1 damage.
2015-09-23 03:21    naomiamerica attacks Vice for 16.2 damage, Vice attacks party for 0.0 damage.
==========
2015-09-22 17:07    Sarah (Diyara) attacks Vice for 5.4 damage, Vice attacks party for 7.2 damage.
2015-09-22 05:33    Sarah (Diyara) attacks Vice for 2.1 damage, Vice attacks party for 3.7 damage.
==========
2015-09-21 19:58    Anna Kettner attacks Vice for 25.8 damage, Vice attacks party for 0.0 damage.
2015-09-21 15:04    naomiamerica attacks Vice for 14.3 damage, Vice attacks party for 0.0 damage.
2015-09-21 07:22    Anna Kettner attacks Vice for 0.0 damage, Vice attacks party for 39.7 damage.
==========
2015-09-20 23:58    naomiamerica attacks Vice for 7.5 damage, Vice attacks party for 0.0 damage.
2015-09-20 16:06    Sarah (Diyara) attacks Vice for 2.9 damage, Vice attacks party for 3.6 damage.
2015-09-20 06:49    naomiamerica attacks Vice for 9.6 damage, Vice attacks party for 0.0 damage.
2015-09-20 04:57    Sarah (Diyara) attacks Vice for 6.0 damage, Vice attacks party for 0.0 damage.
==========
2015-09-19 09:04    Anna Kettner attacks Vice for 10.2 damage, Vice attacks party for 30.0 damage.
2015-09-19 07:27    naomiamerica attacks Vice for 16.7 damage, Vice attacks party for 0.0 damage.
2015-09-19 06:36    Anna Kettner attacks Vice for 20.9 damage, Vice attacks party for 4.3 damage.
==========
2015-09-18 16:42    Sarah (Diyara) attacks Vice for 7.3 damage, Vice attacks party for 0.0 damage.
2015-09-18 03:08    Mike (Korban) attacks Vice for 1.5 damage, Vice attacks party for 17.2 damage.
==========
2015-09-17 17:25    Sarah (Diyara) attacks Vice for 7.4 damage, Vice attacks party for 0.0 damage.
2015-09-17 15:00    naomiamerica attacks Vice for 14.6 damage, Vice attacks party for 0.0 damage.
2015-09-17 14:37    Anna Kettner attacks Vice for 20.7 damage, Vice attacks party for 7.6 damage.
2015-09-17 07:03    Anna Kettner attacks Vice for 13.0 damage, Vice attacks party for 24.1 damage.
2015-09-17 00:42    Mike (Korban) attacks Vice for 8.7 damage, Vice attacks party for 4.2 damage.
==========
2015-09-16 16:49    Sarah (Diyara) attacks Vice for 8.0 damage, Vice attacks party for 0.0 damage.
2015-09-16 15:30    naomiamerica attacks Vice for 8.3 damage, Vice attacks party for 0.0 damage.
2015-09-16 06:43    Anna Kettner attacks Vice for 21.2 damage, Vice attacks party for 0.0 damage.
2015-09-16 04:53    Sarah (Diyara) attacks Vice for 32.4 damage, Vice attacks party for 0.0 damage.
==========
2015-09-15 22:20    Mike (Korban) attacks Vice for 25.4 damage, Vice attacks party for 0.0 damage.
2015-09-15 16:46    naomiamerica attacks Vice for 10.5 damage, Vice attacks party for 0.0 damage.
2015-09-15 01:15    naomiamerica attacks Vice for 7.5 damage, Vice attacks party for 0.0 damage.
2015-09-15 00:17    Mike (Korban) attacks Vice for 7.9 damage, Vice attacks party for 4.2 damage.
==========
2015-09-14 16:15    Anna Kettner attacks Vice for 7.5 damage, Vice attacks party for 21.2 damage.
2015-09-14 10:29    naomiamerica attacks Vice for 7.5 damage, Vice attacks party for 0.0 damage.
==========
2015-09-13 20:42    Anna Kettner attacks Vice for 12.3 damage, Vice attacks party for 18.0 damage.
2015-09-13 15:21    Mike (Korban) attacks Vice for 8.7 damage, Vice attacks party for 3.2 damage.
2015-09-13 02:06    naomiamerica attacks Vice for 10.1 damage, Vice attacks party for 0.0 damage.
2015-09-13 02:04    Mike (Korban) attacks Vice for 4.8 damage, Vice attacks party for 10.5 damage.
==========
2015-09-12 22:50    Anna Kettner attacks Vice for 24.0 damage, Vice attacks party for 0.0 damage.
2015-09-12 03:52    naomiamerica attacks Vice for 9.8 damage, Vice attacks party for 0.0 damage.
==========
2015-09-11 22:30    Mike (Korban) attacks Vice for 9.8 damage, Vice attacks party for 3.1 damage.
2015-09-11 08:30    Anna Kettner attacks Vice for 19.7 damage, Vice attacks party for 6.1 damage.
2015-09-11 00:29    Mike (Korban) attacks Vice for 14.5 damage, Vice attacks party for 6.1 damage.
2015-09-11 00:29    naomiamerica attacks Vice for 8.2 damage, Vice attacks party for 0.0 damage.
```

**Comment by mikebranski** [comment]
@Alys I did some more digging on https://github.com/HabitRPG/habitrpg/issues/2805#issuecomment-143379666. I use multiple devices, and each of them have identical timestamps, so that doesn't seem to be the issue. The weirdest thing for me is the seemingly random times our crons are being run.

**Comment by TetoIsGreat** [comment]
Hey I am writing to say that my party has been dealing with this issue over and over for the last few months.  It has happened to at least three of us, including me. Last week it happened and killed us all, it just happened again this morning to the same party member.  I do use both the iphone app and the desktop version.  I check in first thing in the morning on the app usually.  I often have Habitica open in more than one tab, but I usually put the computer to sleep at night. When it happened to me it was a few months back, before I switched to the new native app. I have asked the other members of my party to comment in case they have more insight, especially the person being hit with it lately.

**Comment by jchiasson** [comment]
This is happening to me as well. It seems to happen mostly when I sign in from my phone but it doesn't always happen.

**Comment by ghost** [comment]
I am also having this issue. Damage is identical. 
![new bitmap image](https://cloud.githubusercontent.com/assets/8877144/10977356/9419d9ea-83be-11e5-9b91-0dd92ddca1c6.jpg)
I have a custom cron and use two devices (PC and iPhone app), though both are in the same time zone. I think @brandonlichtenwalner is on to something. I use the app at night and sometimes don't close the app until the next day, after I've been on my desktop for a while. This may explain the seemingly random double damage I do to quest bosses.

**Comment by ghost** [comment]
Same issue occurred today under similar circumstances as my last post. Before cron, I last checked Habitica on my laptop. After cron, I first checked Habitica on my iPhone. Upon opening the app, it refreshed info on Habits, Dailies, and To-Dos, but not Party. I had to go to Menu-->Party and refresh that page separately. Double damage appeared. I have the most recent version of the app.

**Comment by Alys** [comment]
@lmking How much time do you estimate passed between the fresh information about your tasks appearing, and you refreshing the party page? What is your User ID (from Settings -> API on the website, but it will be in the app somewhere too).

**Comment by ghost** [comment]
The time gap was short—maybe 30 seconds. My User ID is 88357773-a371-41f5-9ee2-5c4def77a810 .

**Comment by Alys** [comment]
For @lmking's two reports, the duplicate damage occurred within 0.2 seconds in the first case, and 1.9 seconds in the second. 

```
        {
            "id": "2e130934-b1c1-4ea4-9ef5-e2a820707aaa",
            "text": "`Lindsey attacks Rooster for 17.1 damage, Rooster attacks party for 1.3 damage.`",
            "timestamp": 1446644962452
        },
        {
            "id": "88d0390e-b890-488c-9b11-79a23cc1431c",
            "text": "`Lindsey attacks Rooster for 17.1 damage, Rooster attacks party for 1.3 damage.`",
            "timestamp": 1446644962246
        },

0.206 seconds difference between timestamps


        {
            "id": "c7ec977c-28d6-4de9-a197-81b72458c4b3",
            "text": "`Lindsey attacks Octothulu for 16.0 damage, Octothulu attacks party for 0.0 damage.`",
            "timestamp": 1447488549821
        },
        {
            "id": "70f86046-15cf-4464-8dfe-de14cd4f4d4b",
            "text": "`Lindsey attacks Octothulu for 16.0 damage, Octothulu attacks party for 0.0 damage.`",
            "timestamp": 1447488547881
        },

1.940 seconds difference between timestamps

```

**Comment by ohmissjamie** [comment]
Hello! A party member has encouraged me to submit a bug here, I hope this is the right place to do so! 

Reading through the comments a bit, I see that I am not the only one experiencing the "double damage bug". Every night I attack twice (or am attacked twice, depending). Whatever damage I deal and am dealt, it happens twice every night consistently.  It was suggested that I would be a good test case since it's happening every night to me. 

I have a custom cron (3am) and use habitprg on my laptop, and on my android phone (same time zone).

![habitrpg](https://cloud.githubusercontent.com/assets/15874749/11188471/dc40594a-8c3f-11e5-9816-c0045bc10b43.JPG)

Thanks in advance for your time and help. Let me know if there is any other info or screen shots you need!  -- OhMissJamie

**Comment by Alys** [comment]
@ohmissjamie You are right that you would be a good test case! If you don't object, one night when you are on a quest, could you please log out of the mobile app (that will stop it taking any actions on your account that we aren't aware of, such as background syncs), and also close all browser tabs that have Habitica open, on all laptops or PCs that you use Habitica on. 

If you use any third-party tools, please also disable them. Feel free to ask for clarification about that, although if you don't know of any third party tools, this almost certainly won't be relevant to you.

Then the next morning, please open Habitica in one browser tab only. Don't open it in any other tabs, or use any third party tools, or log in to the Android app, until you have seen that your Dailies have reset. Then please tell us whether you still do double damage!

I realise all of this might be inconvenient and might conflict with your preferred workflow, so if you'd rather not do this that's quite okay! But it would be very interesting test.

**Comment by ghost** [comment]
![habitica2damage](https://cloud.githubusercontent.com/assets/8877144/11344324/7ea83fa6-91de-11e5-8fe1-6be991fb8d64.jpg)

Happened again. I've been logging when the problem occurs. It seems to be a problem with the app, not the website. When I leave the app open before cron and open the app the next morning, double damage. If I exit the app before cron (just swiping up to close, not logging out), I never inflict double damage. I don't use third party tools, and I never keep browser tabs open after I'm done on my computer for the night. Next time, I'll try leaving the app open before cron, then using the website first rather than the app after cron.

**Comment by gaballison** [comment]
It's still happening to me in conjunction with not being able to log in for 3-4 tries a day. I've only used the Habitica app once or twice in the last month so I don't think my issues are app-related. You can see here that the amount of damage I do or take varies considerably:
![habitica-cropped](https://cloud.githubusercontent.com/assets/12204283/11504804/101708f0-9815-11e5-92a7-816f9e6cd748.png)
(Green text is normal, yellow is duplicate damage)

Several weeks ago I had an incident where it took me 3-4 tries to log in, after not being able to log in at all the day before (or rather giving up after the 3rd try) and therefore our party was dealt 4x damage:
![habitica-4xdamage](https://cloud.githubusercontent.com/assets/12204283/11504840/4f13aa04-9815-11e5-9504-3af18f13514b.png)

**Comment by SabreCat** [comment]
@gaballison: What does it look like when you're not able to log in? Does it tell you your password isn't recognized, or display some other error?

**Comment by lemoness** [comment]
This happened to me this morning. Salient details - I accidentally had Habitica open in two different windows, and checked off tasks on both of them before I realized. When I refreshed to sync them up, I got the popup telling me that the Cobra boss had been defeated (by my second attack).

**Comment by gaballison** [comment]
@SabreCat When I can't log in, it allows me to enter my username and password and the horizontal loading bar at the top of the page goes completely across, which you would think would mean I'm logged in, but then I'm not redirected to my tasks screen. I try either hitting the submit button again or refreshing the page and it still keeps me on the front page. Sometimes it takes several tries of refreshing or resubmitting before it will let me in.

The logging in issue is usually after I've run CCleaner and deleted all of my cookies, etc. so I have to re-log in everywhere again. More often my problem is when I'm already signed in and my dailies don't reset when I open Habitica in the morning, and then I do get application error messages. In those cases I also have to refresh 2-4 times before everything finally resets, and sometimes (although not all the time) that results in this multiple damage bug.

**Comment by SabreCat** [comment]
@gaballison: OK, got it. Next time you do a cookie cleanup, you could try deleting local storage for habitica.com as well--that might make the login process go more smoothly.

**Comment by lemoness** [comment]
This happened last night. Some details:
1. I normally use Habitica on more than one device, but actually I haven't right now, because I was testing new iOS builds, so my standard Lemoness account has not been logged in to the mobile app for several days. In fact, currently I'm not logged in to any account on mobile at all.
2. The damage was identical, although I am not sure if it was applied twice or merely reported twice. 
3. Left Habitica + DDT open overnight as always, but not in more than one window/browser.
   First thing I did in the morning was refresh Habitica, then Re-Fetch data from the DDT, in quick succession. (One might not have been loading before I did the other.) Then I checked off a couple of morning Dailies, opened party chat, and saw the double damage.

**Comment by Alys** [comment]
I've confirmed that lemoness's damage was applied to the quest twice, not merely reported in chat twice (all damage messages from that quest, including the duplicate, added up to the total health loss of the boss).

**Comment by brightbold** [comment]
Well after not having had this happen to me in ages, I've done triple damage the last two days in a row. I typically keep three Habitica tabs open in Chrome, but in the past this hadn't caused a problem unless I make a second (or third) Habitica tab active while the first one is loading. If I actively cause more than one tab to load at once, I can trigger this bug. So I don't do that! I wonder what has changed in the last two days (about Habitica? Or Chrome? Or me?) that's causing this to operate differently. (Assuming my theory is correct that the triple damage is due to my three tabs, and not just a coincidence...)
![habitica-triple-1](https://cloud.githubusercontent.com/assets/405068/13904711/97cee676-ee7f-11e5-9cc2-b7ec862ae687.png)
![habitica-triple-2](https://cloud.githubusercontent.com/assets/405068/13904712/995fc6e0-ee7f-11e5-98ac-a391071f6207.png)

**Comment by Alys** [comment]
@brightbold It sounds like when you make two or three tabs running Habitica active at the same time, cron starts running in each of them. It wouldn't necessarily happy every time because if cron in one tab can complete before it starts in the next, then it wouldn't actually start in the next. The past two days might have been different because network conditions were ideal for all three tabs to contact the server simultaneously. 

Or are you saying that you did NOT make all three tabs active at the same time over the past couple of days?

**Comment by brightbold** [comment]
@Alys I did NOT make all three tabs active at the same time. In my experience, doing that would definitely trigger multiple simultaneous crons. 

Even though I have Chrome set to re-open all my tabs from the previous session, normally Chrome will only load the tab if you click on it to make it active, so I'm careful to wait until one Habitica tab has loaded before clicking another one. So I thought maybe Chrome had changed something that broke that functionality, or that something changed on Habitica's end, causing all three tabs to load simultaneously. But today cron only ran once, so I don't know what might have been different the last two days.

**Comment by Alys** [comment]
@brightbold Habitica does have a feature that causes the website to reload itself if it's been inactive for six hours. Maybe that kicked in on all the open tabs on those two days. Other than that, I don't have any ideas at the moment.

**Comment by lemoness** [comment]
This happened with one of my API test accounts, but not the other.
- Account was created yesterday. Tab was not left open overnight anywhere on my computer. Account WAS logged in on mobile phone, but had not been checked since the previous day. 
- Account dealt damage (in addition to having it reported) a total of three times.

**Comment by Pekyy** [comment]
Hi, this bug occured for the first time in our party. I think, it happened because of the maintenance. Tripled and doubled damage for 4 our members. I use Habitica via webapp and ios app. I also login from different computers. Damage is identical.
![bez nazvu](https://cloud.githubusercontent.com/assets/18715151/15464488/f7841c26-20cf-11e6-87fd-8cff118e6900.jpg)
![bez nazvu2](https://cloud.githubusercontent.com/assets/18715151/15464486/f783b024-20cf-11e6-8e55-7837cb898c39.jpg)
![bez nazvu3](https://cloud.githubusercontent.com/assets/18715151/15464487/f783f2aa-20cf-11e6-99c2-600ee324f081.jpg)

**Comment by Alys** [comment]
Another report of this here: https://github.com/HabitRPG/habitrpg/issues/7436#issuecomment-220951700
It happened three times in three days.

@khipkin Are you using the website and a mobile app? Do you have the website open in multiple tabs or windows each day?

**Comment by khipkin** [comment]
I use the website most often, but I use the Android app if I'm away from my computer.  The last few days, I've had two extra tabs open (in addition to the main one that I keep pinned).

Also, it only happened twice in two days.  That's why I thought it might have been because of the migration.

**Comment by Alys** [comment]
@khipkin Ah, sorry, twice. I both misread and miscounted. :-/ Thanks for all the information!

**Comment by LlamaHobbit** [comment]
This hasn't happened to me in several months. But happened again this morning. All tabs were closed on both my PC and my Android phone last night before bed. Logged in with a new tab in Chrome this morning, and it happened again. But it didn't seem to happen immediately. I went to the Party chat via the Notifications tab to see how much damage I'd done and ONLY showed the boss damage once. I navigated away to another guild's chat, then saw that I had another Party notification. Back to party chat, but only saw my previous one-time damage. I clicked on "Fetch Recent Messages" and that is when my second damage showed up along with a corresponding second drop in the Boss health. Also, even though there were an intervening few minutes between when the two damage notifications showed up, they both show the time from when the second damage occurred. 

![double damage](https://cloud.githubusercontent.com/assets/7524843/15470531/d310ecc0-20b6-11e6-95d3-57e95eadbe41.PNG)

And while I was typing that up ... it happened to the next party mate who logged in. Our party ID is : ab82c2b2-5f26-4ac2-b7ff-ee940ea050a6

![dd 2](https://cloud.githubusercontent.com/assets/7524843/15470823/5f333acc-20b8-11e6-899c-4fe894e646da.PNG)

**Comment by Alys** [comment]
For @LlamaHobbit's attacks, the timestamps are nearly identical:

```
1464003206946 = 2016-05-23 11:33:26 UTC
1464003206997 = 2016-05-23 11:33:26 UTC
```

Same applies to the next pair of damage messages too.

The delay in the messages appearing might have been from the odd chat syncing we've been noticing.

**Comment by crookedneighbor** [comment]
Can we confirm that the latests instances of double quest progress have _actually_ applied damage? Or is it just a matter of duplicating the message?

**Comment by Alys** [comment]
Boss health initially 500, total damage 487.4, remaining health 12.6, so all of the damage is being applied.

"text": "`breadmaker attacks Skeletal Tyrannosaur for 36.6 damage.` 
"text": "`breadmaker attacks Skeletal Tyrannosaur for 36.6 damage.` 
"text": "`Llama Hobbit attacks Skeletal Tyrannosaur for 192.9 damage.` 
"text": "`Llama Hobbit attacks Skeletal Tyrannosaur for 192.9 damage.` 
"text": "`Sir Antrue attacks Skeletal Tyrannosaur for 3.7 damage.` 
"text": "`MEmry, the silver pirate attacks Skeletal Tyrannosaur for 5.8 damage.`
"text": "`breadmaker attacks Skeletal Tyrannosaur for 10.3 damage.` 
"text": "`Sir Antrue attacks Skeletal Tyrannosaur for 8.6 damage.`

**Comment by ltorroba** [comment]
This bug has just happened to me. I use Habitica in multiple devices (and sometimes multiple tabs), but today I have not touched my phone and I have only one tab open. Both clocks are set to the same timezone as my Habitica account.

Interestingly enough, we are doing a collection quest, not a boss quest.

![](http://i.imgur.com/29moErx.png)

I had never noticed this issue until now. Other party members did not exhibit this problem.

UUID: 8ac1be04-4201-4eed-a2a4-436eada5c0e9

**Comment by ReveWeber** [comment]
We have had our first experience with this as a party in the past two days - our warrior did double damage yesterday and triple damage today. It is definitely applying to the boss's health. He says he hasn't noticed anything else strange (eg double mana), he is pretty sure he had only one tab open, and while he may have used the iOS app in the past, the last occurrence of it would have been months ago. When I mouse over the timestamp the duplicates are identical down to the second.
<img width="729" alt="screen shot 2016-05-23 at 6 56 05 pm" src="https://cloud.githubusercontent.com/assets/11982927/15487985/b07fc8fc-211f-11e6-85de-86449e8b493a.png">
<img width="800" alt="screen shot 2016-05-23 at 6 56 30 pm" src="https://cloud.githubusercontent.com/assets/11982927/15487986/b08210da-211f-11e6-9d9b-7c139269b9f5.png">

His UUID is 5772e6a4-932b-468d-90f2-ecbaa5afd06d
Our group ID is 37dc23f3-4af2-4024-9e38-1254a359723b

**Comment by LlamaHobbit** [comment]
@Alys This might be the same as what you meant on the timestamps. But just to be sure, what I mean is that I saw the first damage, and saw the corresponding drop in boss health by 196, and saw the message stamped as having happened "X minutes ago" where X was more than 1 (don't remember the exact number). When I navigated back to the page, and hit Fetch Recent Messages, then the second damage message occurred, the corresponding second drop in health happened, _and the timestamp from the first one changed_. So that the first timestamp appeared to update to match the correct timestamp for the new second one - now both read "1 minute ago." When the first had already read as _more_ than 1 minute ago previously.

**Comment by Alys** [comment]
@LlamaHobbit Ohhhhh... yes, I had missed that. Thank you for explaining again! That's very odd. At the moment, I really don't know what happened there.

@zesme @ReveWeber Thanks for your detailed reports! It's definitely helpful to know that this happened with just one tab open.

**Comment by Bottl** [comment]
Hello again Alys, just letting u know it's still happening in our party too

**Comment by ghost** [comment]
Adding my report: Triple damage done (all a at once). I accessed Habitica on my iPad using Safari to trigger cron. The Habituca app (updated after maintenance) was also open (I was trying out the reminders function) but I hadn't used the app since yesterday morning. Habitica is open on my desktop but that computer is idle at night and shouldn't trigger anything (and never has before). The app running in the background on the iPad may or may not be relevant and is the only new variable other than maintenance. 
![image](https://cloud.githubusercontent.com/assets/6047755/15503429/aed264d4-216e-11e6-9fc5-d9b83a63d66e.jpeg)
Edit: screenshot cropped to remove some private party info

**Comment by Gooseus** [comment]
I just started having this issue at the last CRON.

![image](https://cloud.githubusercontent.com/assets/39137/15517672/dd631f48-21ad-11e6-8e65-1658a73e655f.png)

I can confirm that the damage definitely double applied from me.

While I see this is a long running issue, I'll put my theory up anyways.

Prior to the update I had a persisting bug with still being in the invited state for the party, even though I was also in the party.  I was in the Invited list and I had the confirmation at the top to Accept/Reject the invitation... though no matter what I input to that it would always be there.

After the update my fellow party members had no issues, but I seemed to not be in any party... though they assured me I was still in the party from their point of view.

I reported the issue and within a day I saw I was back in the party, though it was asking me to accept the Quest that I had started the day prior to the update.  I accepted and everything was fine, I even think that the CRON ran fine that day and I only attacked once.

That was yesterday, and then today I saw that above screenshot attacking twice and I also noticed that I was no longer in the Invites list.

My theory was somehow I had been added into the group twice and that's why an attack was coming from me twice.

Take that for what you will.  Based on the number of times this has been reported, I assume this would have been noticed if it was simple as people being added to groups multiple times.

**Comment by SabreCat** [comment]
Possible fix underway at #7485 and 91e16ff1910240e7b740aeeeb1b11ef770211393

**Comment by lemoness** [comment]
We've just deployed a fix that should prevent most of these cases! We are
still working on a fix to eliminate them completely, but they should become
much less frequent now.

**Comment by ShulamitLH** [comment]
I just saw it again today and yesterday, both times for the same player (Shulamit). 

Shulinde attacks Vice, the Shadow Wyrm for 24.8 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon! -3 hours ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 49.0 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon! -3 hours ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 49.0 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon! -3 hours ago+1   

Athelinde attacks Vice, the Shadow Wyrm for 109.5 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon! -3 hours ago+1   

Athelinde attacks Vice, the Shadow Wyrm for 82.8 damage. Vice, the Shadow Wyrm did not attack the party because it was asleep while maintenance was happening. -a day ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 107.7 damage. Vice, the Shadow Wyrm did not attack the party because it was asleep while maintenance was happening. -a day ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 107.7 damage. Vice, the Shadow Wyrm did not attack the party because it was asleep while maintenance was happening. -a day ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 107.7 damage. Vice, the Shadow Wyrm did not attack the party because it was asleep while maintenance was happening. -a day ago+1   

Shulinde attacks Vice, the Shadow Wyrm for 70.0 damage. Vice, the Shadow Wyrm did not attack the party because it was asleep while maintenance was happening. -a day ago+1 

In our party, 3 of us are battling the boss. The double/tripple damage was only with Shulamit.

**Comment by paglias** [comment]
Shulamit could you try clearing the cache and then reloading Habitica?
Il 25 mag 2016 6:39 AM, "ShulamitLH" notifications@github.com ha scritto:

**Comment by Alys** [comment]
The exact timestamp for Shulamit's most recent double-cron (49.0 damage) is 2016-05-25 01:56:19 (about 7.5 hours ago).

**Comment by Alys** [comment]
Ignoring Shulamit's report just for the moment: I've checked the quest progress messages for all those parties reported here recently and also all the Tavern reports today, and there's been no duplicate damage since the two fixes several hours ago. If we don't see any others in the next maybe 12 hours, then I'd think that this current run of this bug has been fixed by removing the new extra sync and cron call.

Shulamit's report might be a coincidental example of the original bug. It wasn't happening very often but often enough that the coincidence isn't unbelievable.

**Comment by Memry** [comment]
UUID: 2dc2722b-fdc1-4a36-894f-c22418b67a30
Browser: Opera
OS: Windows 7

I have my cron set to 6 am.  I had my account open on one tab only and was working in a different browser when 6 am came around. Before 6 am  I had looked at my account and saw my due dailies were all checked.  Our party is curently in a quest, so after 6 am I checked on the task page and all the to dos showed unchecked as expected, I check in the party and it did not show my damage to the boss.  I refreshed.  It showed double damage to the boss and all my streaks on the due dailies reset to 0.

**Comment by Alys** [comment]
Memry, did you lose any health?

**Comment by Memry** [comment]
I did.  That's what caught my attention first was the message about losing health.

**Comment by paglias** [comment]
@Memry can you try clearing the cache and then reloading Habitica?

**Comment by paglias** [comment]
It won't solve your double cron but may prevent it in the next days

**Comment by Memry** [comment]
I have cleared the cache and reloaded.  We will see what happens tomorrow morning.

**Comment by ShulamitLH** [comment]
Re: [HabitRPG/habitrpg] Quest damage ran twice - doubled boss/collection quest progress (#2805)

Hello Matteo,

I tried that. It looks like I didn't do double damage, but another of my party did instead.

Athelinde attacks Vice, the Shadow Wyrm for 55.3 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon!
-2 hours ago+1   

Athelinde attacks Vice, the Shadow Wyrm for 55.3 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon!
-2 hours ago+1   

Shulinde attacks Vice, the Shadow Wyrm for 53.8 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon!
-3 hours ago+1   

Shulamit attacks Vice, the Shadow Wyrm for 51.4 damage. Vice, the Shadow Wyrm does not attack, because it respects the fact that there are some bugs post-maintenance and it doesn't want to hurt anyone unfairly. It will continue its rampage soon!

Wednesday, May 25, 2016, 11:07:35 AM, you wrote:

Shulamit could you try clearing the cache and then reloading Habitica?
Il 25 mag 2016 6:39 AM, "ShulamitLH" notifications@github.com ha scritto:

**Comment by Memry** [comment]
Today looked more normal for me.

**Comment by lemoness** [comment]
We've added a change that should have fixed this! If anyone notices the
quest doubling bug happening to them, please let us know as soon as
possible.

**Comment by Alys** [comment]
https://github.com/HabitRPG/habitrpg/issues/7527 might be an example of this bug

**Comment by Squeemu** [comment]
I commented in another thread, but I see this is probably the better one, so I'll just copy and paste my other response.

I think I'm having the same issues: streaks were reset on my Friday dailies and Perfect Day bonus lost, even though I checked all my dailies off.

I'm currently on a quest and it appears that I cronned twice, both times appearing as "an hour ago." The first cron shows that I did damage to the boss and took no damage, the second one shows that I did zero damage but took damage from the boss.

I only use Safari -- no apps. I also only had one Habitica tab open.

UID: df5a0dda-0b53-45a2-9366-d239aa6663bf
Browser: Safari (Version 8.0.5 (10600.5.17))
OS: X Yosemite 10.10.3

<img width="588" alt="screen shot 2016-05-28 at 12 18 48 pm" src="https://cloud.githubusercontent.com/assets/19624863/15628125/dcb298f6-24cf-11e6-91f8-1e05fae28014.png">

**Comment by Alys** [comment]
In #7527 there's a second report, from Squeemu (df5a0dda-0b53-45a2-9366-d239aa6663bf) that is similar to the earlier cases of this bug but has differing amounts of damages:

```
        {
            "id": "8ef1d495-4d29-4f10-912e-357290836c47",
            "text": "`Squeemu attacks Water Skull Swarm for 0.0 damage.` `Water Skull Swarm attacks party for 2.8 damage.`",
            "timestamp": 1464443827557,  [2016-05-28 13:57:07 UTC]
            "flagCount": 0
        },
        {
            "id": "eacdc5f8-fd6f-4af8-87d6-3f333a903e5f",
            "text": "`Squeemu attacks Water Skull Swarm for 14.3 damage.` `Water Skull Swarm attacks party for 0.0 damage.`",
            "timestamp": 1464443827195, [2016-05-28 13:57:07 UTC]
            "flagCount": 0
        },
```

**Comment by Alys** [comment]
Another report in the Tavern from Wabisabi (91cb2031-b9cd-4872-94dc-8768abddb1f7): "But the damage sustained wasn't enough to account for the number of levels that members of my party have gone down. And something is screwy; for example, on 5/27/16 at 19:52:58, @SuperBeagle attacked for 114.8 damage and incurred 0, but then immediately attacked for 0 and incurred 24.7. It's almost like cron happened twice?"

**Comment by Alys** [comment]
Suggestion: until we work out what's causing this and stop it happening, we adjust cron so that if a player has done less than 2 points of damage to the boss (e.g., one or two task's worth of damage), then the boss does not do any damage to the party.

It's a bit extreme because sometimes just one or two points of damage could be a legitimate amount for a full day. We could prevent damage to players happening with there's just 0 points of damage done to the boss, but I'm worried about the second cron happening not fast enough to prevent the user ticking a couple of tasks. Opinions?

We could also make it so that the player's own Dailies don't damage the player themself when there's 2 (or 0) points of boss damage done.  Opinions?

If I don't hear any complaints by the time I've finished breakfast, I'll start working on the code and tests.

**Comment by Alys** [comment]
Reported in the Tavern by Pauglos (0a292c31-a7f9-416b-9f3a-a99afd09f11a):

```
    {
        "id": "4d8f57c7-0a7e-463a-8de9-264b7b98c91a",
        "text": "`Pauglos attacks Gold Knight for 0.0 damage.` `Gold Knight attacks party for 12.8 damage.`",
        "timestamp": 1464590130647, [2016-05-30 06:35:30 UTC]
        "flagCount": 0
    },
    {
        "id": "bed00d52-6443-49b5-a691-5d15bf72897c",
        "text": "`Pauglos attacks Gold Knight for 17.9 damage.` `Gold Knight attacks party for 0.0 damage.`",
        "timestamp": 1464590130484,  [2016-05-30 06:35:30 UTC]
        "flagCount": 0
    },
```

**Comment by Alys** [comment]
maxh (991d201d-8307-4117-a3b3-b974adbf2327) in Tavern:
"I seem to have been hit with a double-cron last night, causing the loss of all streaks, 12 HP, and my perfect day buff."

No quest, but confirmed from a Daily's history:

{ "_id" : "4ebf9746-3406-4a35-9308-66653d503b4d", "history" : [ { ... { "date" : 1464321604555, "value" : 91.38817566702585 }, { "date" : 1464525924658, "value" : 90.80836600503427 }, { "date" : 1464602411567, "value" : 91.38817566702585 }, { "date" : 1464602411758, "value" : 90.80836600503427 } ] }

The final two values are 2016-05-30 10:00:11 UTC

**Comment by Alys** [comment]
Another report of this (the first I've seen since my previous entry a month ago):

user's data:

```
"_id": "771456f9-34fa-40e2-b2e2-f14958b7eda0",
"party": {
    "_id": "edaeca30-f3f4-472b-8c40-b7dd2dcf8604"
},
"lastCron": "2016-06-25T09:50:12.536Z",
"_cronSignature": "NOT_RUNNING",
```

party chat:

```
{
    "id": "65568bfa-bc73-4587-a945-c40c791728bb",
    "text": "`DavosSeaworth attacks Adva, the Usurping Mermaid for 11.3 damage.` `Adva, the Usurping Mermaid attacks party for 10.5 damage.`",
    "timestamp": 1466848212826, //  UTC: 2016-06-25 09:50:12
    "flagCount": 0
},
{
    "id": "1cd0e33a-aae6-4aab-b0b6-27d1848ee2ce",
    "text": "`DavosSeaworth attacks Adva, the Usurping Mermaid for 11.3 damage.` `Adva, the Usurping Mermaid attacks party for 0.0 damage.`",
    "timestamp": 1466848211815, //  UTC: 2016-06-25 09:50:11
    "flagCount": 0
},
```

**Comment by Alys** [comment]
Another report of this (the first I've seen since my previous entry in June):

user's data:

```
{
    "_id": "444cf175-e917-4683-a40d-498b09146148",
    "party": {
        "_id": "700adf43-c50b-4113-808c-00002d93d9f1"
    },
    "lastCron": "2016-08-02T17:00:00.749Z",
    "_cronSignature": "NOT_RUNNING"
```

party chat:

```
        {
            "id": "be739fd9-7527-4e71-a2f7-0f031280343c",
            "text": "`Tahtimittari attacks The Necromancer for 0.0 damage.` `The Necromancer attacks party for 7.1 damage.`",
            "timestamp": 1470157200975, //      UTC: 2016-08-02 17:00:00
            "flagCount": 0
        },
        {
            "id": "1a5fdf9f-15c8-418a-9f39-9de2f31c97e1",
            "text": "`Tahtimittari attacks The Necromancer for 12.1 damage.` `The Necromancer attacks party for 0.0 damage.`",
            "timestamp": 1470157199280, //      UTC: 2016-08-02 16:59:59
            "flagCount": 0
        },
```

[Amplitude record, filtered for cron](https://amplitude.com/app/132025/activity/search?userId=444cf175-e917-4683-a40d-498b09146148&eventData=eyJ1dWlkIjoiOGRhZmM1MjYtNThkMi0xMWU2LTk4NmMtMjIwMDBhMTYwZGZhIiwiZXZlbnRJZCI6NTI5MjE3NTQyLCJjbGllbnRFdmVudFRpbWUiOiIyMDE2LTA4LTAyIDE3OjAwOjAxLjMxMTAwMCIsIm1heFNlcnZlclVwbG9hZFRpbWUiOiIyMDE2LTA4LTAyVDE3OjExOjQ5LjE5OFoiLCJjdXJyZW50UGFnZSI6MCwiZGF0ZSI6bnVsbH0)

**Comment by SabreCat** [comment]
Wild. In some cases, the user's `progress` gets reset in between, in others it doesn't.

**Comment by Alys** [comment]
I'm closing this because I believe it was fixed by a cron improvement that @paglias made some time ago. I don't recall seeing any instances of this bug recently.
